hydromt_wflow/wflow.py

```diff
@@ -2948,6 +2958,529 @@ def setup_1dmodel_connection(
+        additional_parameters: dict
+            Dictionary to provide additional paramaters to the model, by default `{
+                "vertical.paddy": {
+                    "h_min": 20, "h_opt": 50, "h_max": 80,
+                }, "h_values": {  # values in a list from h1, h2, h3_high, h3_low, h4
+                    "paddy": [100, 55, -160, -250, -16000], "nonpaddy": [0, -100, -400,
+                    -1000, -15849],
+                },
+            }` The entries under `vertical.paddy` will be added as constant values in
+            the toml file, through the `vertical.paddy.h_min.value = 20` interface. The
+            values are expected to be given in mm, and represent the following: `h_min`
+            is the minimum required water depth for the irrigated rice field, `h_opt`
+            the optimal water depth for the rice field, and `h_max` this water depth
+            when the rice field starts spilling water.
+
+
+        See Also
+        --------
+        workflows.demand.find_paddy workflows.demand.classify_pixels
+
+        Peano, D., Materia, S., Collalti, A., Alessandri, A., Anav, A., Bombelli, A., &
+        Gualdi, S. (2019). Global variability of simulated and observed vegetation
+        growing season. Journal of Geophysical Research: Biogeosciences, 124, 3569–3587.
+        https://doi.org/10.1029/2018JG004881
+        """
+        if len(wflow_thicknesslayers) != len(target_conductivity):
+            raise ValueError(
+                "Lengths of wflow_thicknesslayers and target_conductivity does not "
+                "match"
+            )
+
+        self.logger.info("Preparing irrigation maps.")
+
+        # Extract irrigated area dataset
+        irrigated_area = self.data_catalog.get_rasterdataset(
+            irrigated_area_fn, bbox=self.grid.raster.bounds, buffer=3
+        )
+
+        # Extract landcover dataset (that includes a paddy/rice class)
+        landuse_da = self.data_catalog.get_rasterdataset(
+            landuse_fn, bbox=self.grid.raster.bounds, buffer=3
+        )
+
+        # Get paddy and nonpaddy masks
+        paddy, nonpaddy = workflows.demand.find_paddy(
+            landuse_da=landuse_da,
+            irrigated_area=irrigated_area,
+            paddy_class=paddy_class,
+        )
+
+        # Get paddy and non paddy pixels at model resolution
+        wflow_paddy = workflows.demand.classify_pixels(
+            da_crop=paddy,
+            da_model=self.grid[self._MAPS["basins"]].raster.mask_nodata(),
+            threshold=area_threshold,
+        )
+        wflow_nonpaddy = workflows.demand.classify_pixels(
+            da_crop=nonpaddy,
+            da_model=self.grid[self._MAPS["basins"]].raster.mask_nodata(),
+            threshold=area_threshold,
+        )
+
+        # Add maps to grid
+        self.set_grid(wflow_paddy, name=self._MAPS["paddy_area"])
+        self.set_grid(wflow_nonpaddy, name=self._MAPS["nonpaddy_area"])
+
+        # Update config
```

Comment:
If the wflow model includes paddy fields, the config file should also be updated for the paddy water depth state `h`, e.g.:
```toml
[state.vertical.paddy]
h = "h_paddy"
````

Reply:
Thanks for spotting this; this is now included!